src/Faker/Provider/Base.php

```diff
@@ -110,6 +110,9 @@ public static function randomFloat($nbMaxDecimals = null, $min = 0, $max = null)
 
         if (null === $max) {
             $max = static::randomNumber();
+            if ($min > $max) {
+                $max = $min;
+            }
         }
 
         if ($min > $max) {
```

Comment:
@fzaninotto 

Without the change proposed in this PR, `$min` would then be set to `0`, and eventually return `0`, too, in rare cases.